### PR TITLE
Parse parts and subparts

### LINKED ISSUE #9: Differentiate IEEE Standards vs IEEE Non-standards
IEEE Standards have the pattern "IEEE Std ..." while non-standards are "IEEE ...":

```
IEEE Std 1027
IEEE Stad 204-1961
IEEE Standard No 18-1968 (ANSI C55.1-1968)
```

### PR DESCRIPTION
**Pull request by mico**
related #4 
closes #9

### CODE REVIEW

File: lib/pubid/ieee/identifier.rb [hunk 1]
```diff
@@ -11,11 +11,19 @@ def self.parse(code)
       new(**Parser.new.parse(code).to_h)
 
     rescue Parslet::ParseFailed => failure
-      raise Pubid::Iso::Errors::ParseError, "#{failure.message}\ncause: #{failure.parse_failure_cause.ascii_tree}"
+      raise Pubid::Ieee::Errors::ParseError, "#{failure.message}\ncause: #{failure.parse_failure_cause.ascii_tree}"
     end
 
     def to_s
-      "IEEE #{number}-#{year}"
+      "IEEE Std #{number}#{part}#{subpart}-#{year}"
```
Comment (ronaldtse) [inline lib/pubid/ieee/identifier.rb]: There are Standards vs non-Standards. Only Standards should have the "Std" label.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/pubid/ieee/parser.rb [hunk 2]
```diff
@@ -5,17 +5,53 @@ class Parser < Parslet::Parser
...
+          # 1244-5.2000
+          # 11073-40102-2020
+          # C37.0781-1972
+          (part >> year) |
+          # C57.19.101
+          part >> subpart |
+          # 581.1978
+          year |
+          # 61691-6
+          part
```
Comment (mico) [inline lib/pubid/ieee/parser.rb:53]: @ronaldtse Started with basic identifiers, but already have a challenge to parse them correctly... 😥

**Comment by mico** [comment]
closes #9

**Comment by ronaldtse** [comment]
Thanks @mico !

**Pull request merged**